### PR TITLE
Update crate ed25519-dalek from 1.0 to 2.0

### PR DESCRIPTION
**Pull request by sydhds**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

**Comment by damip** [comment]
@sydhds nice !!
Can you ensure retro-compatibility ? eg. try to verify and sign in both directions between an "old" and a "new" wallet ?

**Comment by sydhds** [comment]
> @sydhds nice !! Can you ensure retro-compatibility ? eg. try to verify and sign in both directions between an "old" and a "new" wallet ?

I've been testing in the following way:
* Checkout branch with ed25519-dalek v2.0.0
* Generate a new address and sign a message using massa-client
* Checkout branch with ed25519-dalek v1.0.0
* Relaunch massa-client, and reuse the previous wallet to sign the same message
* Compare both signatures

then done the same process again but with reversed versions (start with v1 then checkout v2).

All good!

File: Cargo.toml [hunk 1]
```diff
@@ -123,7 +123,8 @@ ctrlc = "3.2.5"
 dialoguer = "0.10"
 directories = "4.0"
 displaydoc = "0.2"
-ed25519-dalek = "=1.0"
+# ed25519-dalek = "=1.0"
+ed25519-dalek = { version = "2.0", features = ["rand_core", "zeroize"] }
```
Comment (damip) [inline Cargo.toml:127]: the version should be fixed "=2.0" like before

File: massa-signature/src/signature_impl.rs [hunk 2]
```diff
@@ -25,16 +25,8 @@ use transition::Versioned;
 #[allow(missing_docs)]
 /// versioned KeyPair used for signature and decryption
 #[transition::versioned(versions("0", "1"))]
```
Comment (damip) [inline massa-signature/src/signature_impl.rs:27]: We should drop the fake versions we created for testing

File: Cargo.toml (hunk 1 above)
Comment (sydhds) [reply Cargo.toml]: Done

File: massa-signature/src/signature_impl.rs (hunk 2 above)
Comment (sydhds) [reply massa-signature/src/signature_impl.rs]: Removed :)

**Review by damip — APPROVED** [review]

**Pull request merged**